lib/dsl/jdl_parser.js

```diff
@@ -571,6 +571,7 @@ module.exports = class JDLParser extends Parser {
   configValue() {
     this.RULE('configValue', () => {
       this.OR([
+        { ALT: () => this.CONSUME(LexerTokens.NAME_CAN_START_DIGIT) },
```

Comment:
I don't like it at all. There's no control over each property, we just say "okay, these types are possible and let the validator check everything's okay".

Reply:
That is basic separation of concerns, Syntax versus Type Checks.
Also implemented like  this in the JDL IDE.
https://github.com/jhipster/jhipster-ide/blob/master/jdl/io.github.jhipster.jdl/src/io/github/jhipster/jdl/JDL.xtext#L74

Its like applying a JSON schema on a JSON input.
You can expand it to the full grammar with the very specific types for each config key.
There is nothing preventing doing so. 

A gentle reminder that you yourself complained that maintaining the old system was a chore.
https://github.com/jhipster/jhipster-core/issues/198#issuecomment-374896968
This way its much easier to add new config keys (you don't even have to change the parser!).

Reply:
Well as Deepu said, aging made me a bit wiser ;)